govcd/catalog.go

```diff
@@ -129,24 +130,48 @@ func (cat *Catalog) UploadOvf(ovaFileName, itemName, description string, uploadP
 		}
 	}
 
-	filesAbsPaths, tmpDir, err := util.Unpack(ovaFileName)
+	isOvf := false
+	fileContentType, err := util.GetFileContentType(ovaFileName)
 	if err != nil {
-		return UploadTask{}, fmt.Errorf("%s. Unpacked files for checking are accessible in: "+tmpDir, err)
+		return UploadTask{}, err
 	}
-
-	ovfFilePath, err := getOvfPath(filesAbsPaths)
-	if err != nil {
-		return UploadTask{}, fmt.Errorf("%s. Unpacked files for checking are accessible in: "+tmpDir, err)
+	if strings.Contains(fileContentType, "text/xml") {
```

Comment:
Few more comments:

- I can't find a check whether the file ends with .ovf to make sure the user provides the right file.
- It may not be clear for the user whether he needs to provide `x.ovf` file inside the folder or is the path to the folder itself enough. I believe we need to clarify where this .ovf file resides (and that it is part of other files in that folder) in changelog and function comment. Because not every user knows that you can upack an OVA file.

Reply:
Thank you.
For item 1, due to file extensions is not always be trusted, especially under linux, user may don't use any file extension. So, I check the file content type to know it's OVF or OVA. Even if user offered an invalid OVF but xml file, upload will return error to complain. It is enough from my view. If you insist the check of .ovf extension is mandatory, I can add it.
For item 2, I've add explanation to changelog and function comment and will push later.

Reply:
I concur you with your approach on item 1. Thanks for explanation!
Added some suggestion on item 2 in-line.

BTW, the two remaining reviewers (@dataclouder and @Didainius) are on vacation currently, so please bare with us.